Counterexamples/AharoniKorman.lean

```diff
@@ -0,0 +1,1188 @@
+theorem not_S_mapsTo_previous (hC : IsChain (· ≤ ·) C)
+    (hCn : (C ∩ level n).Finite) (hn : n ≠ 0)
+    (h : ∀ x ∈ S n C \ (C ∩ level n), f x ∈ C ∩ level (n - 1)) :
+    False := by
+  obtain ⟨a, ha⟩ := (square_subset_S hCn).exists
+  let F : Finset Hollom := (chainBetween a a (3 * a) a).image (embed n)
+  have F_subs : F.toSet ⊆ S n C \ (C ∩ level n) := calc
+    F.toSet = embed n '' chainBetween a a (3 * a) a := Finset.coe_image
+    _ ⊆ embed n '' Finset.Icc (a, a) (3 * a, a) := Set.image_mono chainBetween_subset
+    _ = embed n '' Set.Icc (a, a) (3 * a, a) := by simp
+    _ ⊆ embed n '' Set.Ici (a, a) := Set.image_mono Set.Icc_subset_Ici_self
+    _ ⊆ S n C \ (C ∩ level n) := ha
+  have card_F : #F = 2 * a + 1 := by
+    rw [Finset.card_image_of_injective _ (embed n).injective,
+      card_chainBetween (by omega) (by simp)]
+    omega
+  have F_mapsTo : Set.MapsTo f F {x ∈ C ∩ level (n - 1) | line x < 2 * a} := by
+    intro x hx
+    refine ⟨h _ (F_subs hx), ?_⟩
+    set c := f x with hc
+    have hc' : c ∈ C ∩ level (n - 1) := h _ (F_subs hx)
+    clear_value c
+    rw [coe_image, chainBetween, Ico_self, if_pos (by omega), empty_union, ← Icc_map_sectL] at hx
+    simp only [embed_apply, coe_map, Function.Embedding.sectL_apply, coe_Icc,
+      Set.mem_image, Set.mem_Icc, exists_exists_and_eq_and] at hx
+    obtain ⟨b, ⟨hab, hba⟩, rfl⟩ := hx
+    induction hc'.2 using induction_on_level with | h u v =>
+    simp only [line_toHollom]
+    by_contra!
+    have le : h(b, a, n) ≤ h(u, v, n - 1) := by
+      match n, hn with
+      | n + 1, _ =>
+          simp only [add_tsub_cancel_right]
+          apply HollomOrder.next_add (by omega)
+    have : f h(u, v, n - 1) = f h(b, a, n) := by rw [f.eq_self_of_mem hc'.1, hc]
+    have := le_of_toHollom_le_toHollom (f.eq_of_le this.symm le).ge
+    omega
+  have line_mapsTo :
+      Set.MapsTo line {x ∈ C ∩ level (n - 1) | line x < 2 * a} (Finset.range (2 * a)) := by
+    rintro x ⟨_, hx⟩
+    simpa using hx
+  have F_inj : Set.InjOn f F := f.injOn_of_isChain image_chainBetween_isChain
+  have line_inj : Set.InjOn line {x ∈ C ∩ level (n - 1) | line x < 2 * a} :=
+    line_injOn (n - 1)
+      (hC.mono (by simp +contextual [Set.subset_def]))
+      (by simp +contextual [Set.subset_def])
+  have line_F_mapsTo := line_mapsTo.comp F_mapsTo
+  have := card_le_card_of_injOn _ line_F_mapsTo (line_inj.comp F_inj F_mapsTo)
+  simp only [Finset.card_range] at this
+  omega
+
+/-- The Hollom partial order has no spinal maps. -/
+theorem no_spinalMap (hC : IsChain (· ≤ ·) C) (f : SpinalMap C) : False := by
+  obtain ⟨n, hn, hn'⟩ : ∃ n, n ≠ 0 ∧ (C ∩ Hollom.level n).Finite := by
+    obtain ⟨n, hn, hn'⟩ := Filter.frequently_atTop.1 (Hollom.exists_finite_intersection hC) 1
+    exact ⟨n, by omega, hn'⟩
+  exact Hollom.not_S_mapsTo_previous hC hn' hn (Hollom.S_mapsTo_previous f hC hn)
+
+end Hollom
+
+/--
+The Aharoni–Korman conjecture (sometimes called the fishbone conjecture) says that every partial
```

Comment:
Should we have `def AharoniKorman : Prop :=` to make this clearer?

Reply:
I'm not really sure I see the benefit, but also there's not much harm. It is really that much cleaner to abstract out the contents of the `¬`?